test/test_pytree.py

```diff
@@ -294,25 +305,39 @@ def test_broadcast_to_and_flatten(self):
                     LeafSpec(),
                 ]),
             ]),
-        ]), "L(T(*,*,L(*,*)))"),
-    ], name_fn=lambda _, str_spec: str_spec)
-    def test_pytree_serialize(self, spec, str_spec):
-        self.assertEqual(pytree_to_str(spec), str_spec)
-        self.assertTrue(spec == str_to_pytree(str_spec))
-        self.assertTrue(spec == str_to_pytree(pytree_to_str(spec)))
+        ]),
+    ],)
+    def test_pytree_serialize(self, spec):
+        self.assertTrue(spec == treespec_loads(treespec_dumps(spec)))
 
     def test_pytree_serialize_namedtuple(self):
         Point = namedtuple("Point", ["x", "y"])
         spec = TreeSpec(namedtuple, Point, [LeafSpec(), LeafSpec()])
-        str_spec = "N(Point(x, y),*,*)"
 
-        self.assertEqual(pytree_to_str(spec), str_spec)
-
-        roundtrip_spec = str_to_pytree(pytree_to_str(spec))
+        roundtrip_spec = treespec_loads(treespec_dumps(spec))
         # The context in the namedtuple is different now because we recreated
         # the namedtuple type.
         self.assertEqual(spec.context._fields, roundtrip_spec.context._fields)
 
+    def test_pytree_custom_type_serialize(self):
+        class DummyType:
+            def __init__(self, x, y):
+                self.x = x
+                self.y = y
+
+        _register_pytree_node(
+            DummyType,
+            lambda dummy: ([dummy.x, dummy.y], None),
+            lambda xs, _: Dummy(*xs),
+            to_dumpable_context=lambda context: "moo",
+            from_dumpable_context=lambda dumpable_context: None,
```

Comment:
None is json-dumpable, right? Is it possible to make the context something that is not dumpable?  (a custom NoneObject()?)